service/src/main/java/com/fecklessweasel/service/DepartmentUtil.java

```diff
@@ -0,0 +1,41 @@
+package com.fecklessweasel.service;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.fecklessweasel.service.datatier.SQLSource;
+import com.fecklessweasel.service.datatier.SQLInteractionInterface;
+import com.fecklessweasel.service.objectmodel.CodeContract;
+import com.fecklessweasel.service.objectmodel.ServiceException;
+import com.fecklessweasel.service.objectmodel.ServiceStatus;
+import com.fecklessweasel.service.objectmodel.University;
+import com.fecklessweasel.service.objectmodel.Department;
+
+/**
+ * Utility class for looking up universities from servlet request
+ * @author Anjana Rao
+ */
+public final class DepartmentUtil{
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private static int did;
+    private DepartmentUtil() {}
+    public static Department findDepartment(HttpServletRequest request)
+            throws ServiceException{
+        //create department by looking up request
+        did = Integer.parseInt(request.getParameter("did"));
```

Comment:
No error checking here. What if "did" isn't defined in the request? We will have a NullPointer or ArgumentNull exception when we run parse int and the server will return a crash page to the user.

You should null check request.getParameter("did").
You must also try/catch NumberFormatException from Integer.parseInt() or a user submitted *did* containing letters will crash the page.

You can throw new ServiceException(ServiceStatus.MALFORMED_REQUEST) on these errors to redirect the user to an error page.